repairnator/kubernetes-support/README.md

```diff
@@ -0,0 +1,174 @@
+# Deploy Repairnator in Kubernetes
+Note: to store data you can either deploy activeMQ on cloud described at the bottom of this README or if you have a mongodb outside the cluster you can map it as an external service named as mongo (since all of the default mongo k8s service name is named "mongo" if you don't like a lot changes). More details for how this is done is described in [this link](ttps://cloud.google.com/blog/products/gcp/kubernetes-best-practices-mapping-external-services)
+
+![Design](K8sRepairnatorDesign.jpg)
+
+## Prerequisites
+
+* kubectl
+* a working k8s cluster (preferable high cpu capacity)
+* gcloud (for creating the cluster mentoned above and creating mongodb if you don't have them already)
+
+## Setup ActiveMQ
+
+To deploy ActiveMQ for effective queue managements of jobs to the pipeline and scanner. First apply the yaml file inside the queue-for-buildids folder
+```
+	kubectl create -f /queue-for-buildids/activemq.yaml
+```
+To access to the web interface
+```
+	kubectl get pods 
+```
+It should look like activemq-XXXXXXX-XXXXX. Then 
+```
+	kubectl exec -it activemq-XXXXXXX-XXXXX bash
+```
+To expose to localhost for webinterface access and also publishing(sending message to queue) outside the cluster. 
+```
+	kubectl port-forward activemq-XXXXXXX-XXXXX 8161:8161 61613:61613
+```
+Access webinterface with "http://localhost:8161/admin/" in your browser. The default username is "admin" and password is also "admin".
+
+To send message to queue use the publisher.py script in queue-for-buildids folder. Syntax
+```
+	python publisher.py -d queue-name message-1 message-2 message3 ... message-N
+```
+
+queue-name is according to the format /queue/name. For instance /queue/scanner will send message to the scanner queue.
+You can use this later to manually input build id to the pipeline queue or project to the scanner queue
+
+## Setup scanners and pipelines
+
+To deploy the scanner first go into folder "scanner-dockerimage" then directly build and push the image to docker hub(enviroment variables can be specified or changed later in the "repairnator-scanner.yaml" file. With docker 
+```
+	docker build -t repairnator-scanner:tagname .
+	docker tag repairnator-scanner:tagname YOUR_DOCKERHUB_NAME/repairnator-scanner
```

Comment:
what about using the standard repairnator image instead?

Reply:
Hi the standard image does not have stomp installed unfortunately so it can't communicate with ActiveMQ global queue :) .

Reply:
Could you make a first pull-request to add stomp in the main image then?

Reply:
I think I was a bit unclear with my previous comment :). The problem is that this k8s repairnator image has a script called pipeline-worker.py as the entry point (in the dockerfile) and it executes the pipeline as a subprocess. If we add that as the main image I am afraid that it would break many things since the entry point for the main image is repairnator itself and we wouldn't use activeMQ in the normal case unless it's on cloud.

Reply:
then we want to publish a docker image called `repairnator/repairnator-worker`. WDYT?

Reply:
That would be most desirable in this case :). The RT scanner and maybe some other components are using the docker images to validate git repositories  so I am afraid to make changes there.

Reply:
I will add you to the repairnator organization. What's your Dockerhub identifier?

Reply:
Hi, it's tailp and email is tailp@kth.se :) .

Reply:
I have to configure Docker, this will take some time. For now, please push the image under your Docker account and put your account here instead of YOUR_DOCKERHUB_NAME.

Reply:
here is the name for my own pipeline image. "tailp / repairnator-pipeline" .